lib/mongo_compat_api.js

```diff
@@ -0,0 +1,325 @@
+var Promise = require("bluebird");
+var assert = require("assert");
+
+/**
+ * Implements a MongoDB Node.js 2.0 driver compatible API that works for
+ * both, shared and dedicated database.
+ *
+ * @param ditcher ditcher instance
+ * @param params default params for the current app
+ * @constructor
+ */
+var MongoCompatApi = function (ditcher, params) {
```

Comment:
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) This function has 311 lines, which is greater than the 150 lines authorized. Split it into smaller functions. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AS138)
![MINOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-minor.png) Function has a complexity of 63 which is greater than 20 authorized. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://metrics.skunkhenry.com/coding_rules#rule_key=javascript%3AFunctionComplexity)